python/rikai/spark/sql/codegen/sklearn.py

```diff
@@ -0,0 +1,46 @@
+#  Copyright 2021 Rikai Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import Iterator
+import pandas as pd
+import numpy as np
+
+from pyspark.sql.types import StructType
+from pyspark.sql.functions import pandas_udf
+
+
+def generate_udf(spec: "rikai.spark.sql.codegen.base.ModelSpec"):
+    """Construct a UDF to run sklearn model.
+
+    Parameters
+    ----------
+    spec : ModelSpec
+        the model specifications object
+
+    Returns
+    -------
+    A Spark Pandas UDF.
+    """
+
+    def sklearn_inference_udf(
+        iter: Iterator[pd.Series],
+    ) -> Iterator[pd.Series]:
+        model = spec.load_model()
+        for series in list(iter):
+            assert series.shape == (1,)
+            X = [x.tolist() for x in series[0]]
+            Y = model.predict(X)
```

Comment:
Are we try to predict multiple dependent variables here? The common convention is that X is capitalize because it’s a matrix but y is not because it’s a column vector.